### ISSUE TITLE
Singular dense solve with Method::Dixon with SolutionType::Determinist fails

### ISSUE THREAD

**Issue by Breush** [linbox-team/linbox#189]
With this minimal example, we get `x = [ 0 0 0 ]` as a solution, which is wrong.

```c++
#include <linbox/solutions/solve.h>

using namespace LinBox;

int main(void)
{
    using Ring = Givaro::ZRing<Integer>;
    using Matrix = DenseMatrix<Ring>;
    using Vector = DenseVector<Ring>;

    Ring ZZ;

    Matrix A(ZZ, 1, 3);
    Vector x(ZZ, A.coldim());
    Vector b(ZZ, A.rowdim());
    Ring::Element d;

    A.setEntry(0, 1, 1);
    A.setEntry(0, 2, 2);
    ZZ.assign(b[0], 1);

    // Calling Dixon
    solve(x, d, A, b, Method::Dixon(Method::Dixon::SolutionType::DETERMINIST));

    std::cout << "[ 0 1 2 ] x = [ 1 ] => xNum/xDen = [ " << x[0] << " " << x[1] << " " << x[2] << " ] /" << d << std::endl;

    return 0;
}
```

First investigation shows that `FFPACK::LQUPtoInverseOfFullRankMinor` within `RationalSolver<...>::monolithicSolve` might be the culprit.

**Comment by bdsaunders** [comment]
I'd like to see "determinist" spelled "deterministic".  The former word,
"one who determines" is seldom used and is jarring to the ear in our
context.

**Comment by Breush** [comment]
@bdsaunders All right, I changed "Determinist" to "Deterministic" in my WIP branch #169.

**Comment by jgdumas** [comment]
Solved by fflas-ffpack PR#230
